File: php_SDK/com/payhub/ws/api/RecurringBillResponseInformation.php
<?php

/**
 * Created by PhpStorm.
 * User: agustin
 * Date: 24/07/2015
 * Time: 10:47
 */
class RecurringBillResponseInformation
{
    public $status;
    public $lastRecurringBillResponse;
    public $_links;
    public $errors;
    public $rowData;
    public $metaData;
    private $transactionManager;
    private $billInformation;
    private $cardDataInformation;
    private $customerInformation;
    private $merchantInformation;
    private $scheduleInformation;

    /**
     * @return mixed
     */
    public function getLastRecurringBillResponse()
    {
        return $this->lastRecurringBillResponse;
    }

    /**
     * @param mixed $lastRecurringBillResponse
     */
    public function setLastRecurringBillResponse($lastRecurringBillResponse)
    {
        $this->lastRecurringBillResponse = $lastRecurringBillResponse;
    }

    /**
     * @return mixed
     */
    public function getLinks()
    {
        return $this->_links;
    }

    /**
     * @param mixed $links
     */
    public function setLinks($links)
    {
        $this->_links = $links;
    }

    /**
     * @return mixed
     */
    public function getErrors()
    {
        return $this->errors;
    }

    /**
     * @param mixed $errors
     */
    public function setErrors($errors)
    {
        $this->errors = $errors;
    }

    /**
     * @return mixed
     */
    public function getRowData()
    {
        return $this->rowData;
    }

    /**
     * @param mixed $rowData
     */
    public function setRowData($rowData)
    {
        $this->rowData = $rowData;
    }

    /**
     * @return mixed
     */
    public function getMetaData()
    {
        return $this->metaData;
    }

    /**
     * @param mixed $metaData
     */
    public function setMetaData($metaData)
    {
        $this->metaData = $metaData;
    }

    /**
     * @param mixed $transactionManager
     */
    public function setTransactionManager($transactionManager)
    {
        $this->transactionManager = $transactionManager;
    }

    /**
     * @return mixed
     */
    public function getCardDataInformation()
    {
        if($this->cardDataInformation==null){
            $c = new CardDataInformation($this->transactionManager);
            $c->getDataByTransaction(TransactionType::RecurringBill, $this->lastRecurringBillResponse->getRecurringBillId());
            $this->cardDataInformation=$c;
        }
        return $this->cardDataInformation;
    }

    /**
     * @return mixed
     */
    public function getCustomerInformation()
    {
        if($this->customerInformation==null){
            $c = new CustomerInformation($this->transactionManager);
            $c->setUrl($this->transactionManager->getUrl()."recurring-bill/");
            $c->getDataByTransaction(TransactionType::RecurringBill, $this->lastRecurringBillResponse->getRecurringBillId());
            $this->customerInformation=$c;
        }
        return $this->customerInformation;
    }

    public function getBillInformation()
    {
        if($this->billInformation==null){
            $b = new BillInformation($this->transactionManager);
            $b->setUrl($this->transactionManager->getUrl()."recurring-bill/");
            $b->getBillForRecurringBillInformationById($this->lastRecurringBillResponse->getRecurringBillId());
            $this->billInformation=$b;
        }
        return $this->billInformation;
    }

    /**
     * @return mixed
     */
    public function getMerchantInformation()
    {
        if($this->merchantInformation==null){
            $m = new MerchantInformation($this->transactionManager);
            $m->getDataByTransaction(TransactionType::RecurringBill, $this->lastRecurringBillResponse->getRecurringBillId());
            $this->merchantInformation=$m;
        }
        return $this->merchantInformation;
    }
    public function getScheduleInformation()
    {
        if($this->scheduleInformation==null){
            $s = new ScheduleInformation($this->transactionManager);
            $s->getDataByTransaction(TransactionType::RecurringBill, $this->lastRecurringBillResponse->getRecurringBillId());
            $this->scheduleInformation=$s;
        }
        return $this->scheduleInformation;
    }
    /**
     * @return mixed
     */
    public function getStatusInformation()
    {
        if($this->status==null){
            $s = new StatusInformation($this->transactionManager);
            $s->getDataByTransaction(TransactionType::RecurringBill, $this->lastRecurringBillResponse->getRecurringBillId());
            $this->status=$s;
        }
        return $this->status;
    }
    public static function fromArray($data){
        $recurringBill = new RecurringBillResponseInformation();

        foreach ($data as $key => $value){
            if( property_exists( get_class($recurringBill), $key ) ) {
                if($key=="lastRecurringBillResponse"){
                    $recurringBill->{$key}=RecurringBillResponse::fromArray($value);
                }elseif($key=="errors"){
                    $recurringBill->{$key}=Errors::fromArray($value);
                }elseif($key=="status"){
                    $recurringBill->{$key}=Status::fromArray($value);
                }else{
                    $recurringBill->{$key} = $value;
                }
            }
        }
        return $recurringBill;
    }
}

File: java_SDK/PayHubSDK/src/main/java/com/payhub/ws/api/RecurringBillResponseInformation.java
package com.payhub.ws.api;
import java.io.IOException;
import java.util.List;

import com.payhub.ws.model.Bill;
import com.payhub.ws.model.CardData;
import com.payhub.ws.model.Customer;
import com.payhub.ws.model.Merchant;
import com.payhub.ws.model.RecurringBillResponse;
import com.payhub.ws.model.Schedule;
import com.payhub.ws.model.Status;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.core.JsonParseException;
import com.fasterxml.jackson.databind.JsonMappingException;

@JsonInclude(JsonInclude.Include.NON_NULL)  
public class RecurringBillResponseInformation {
	private Merchant merchant;
	private Customer customer;
	private CardData card_data;
	private Bill bill;
	private Schedule schedule;
	private StatusInformation status;
	private RecurringBillResponse lastRecurringBillResponse;         
	public Object _links;
	private List<Errors> errors;
	public String rowData;
	private Object metaData;
    private TransactionManager transactionManager;
    private BillInformation billInformation;
    private CardDataInformation cardDataInformation;
    private CustomerInformation customerInformation;
    private MerchantInformation merchantInformation; 
    private ScheduleInformation scheduleInformation;
    
	public Merchant getMerchant() {
		return merchant;
	}
	public void setMerchant(Merchant merchant) {
		this.merchant = merchant;
	}
	public Customer getCustomer() {
		return customer;
	}
	public void setCustomer(Customer customer) {
		this.customer = customer;
	}
	public CardData getCard_data() {
		return card_data;
	}
	public void setCard_data(CardData card_data) {
		this.card_data = card_data;
	}
	public Bill getBill() {
		return bill;
	}
	public void setBill(Bill bill) {
		this.bill = bill;
	}
	public Schedule getSchedule() {
		return schedule;
	}
	public void setSchedule(Schedule schedule) {
		this.schedule = schedule;
	}
	public String getMetaData() {
		return (String) metaData;
	}
	public void setMetaData(Object metadata) {
		this.metaData = metadata;
	}
	public RecurringBillResponse getLastRecurringBillResponse() {
		return lastRecurringBillResponse;
	}
	public void setLastRecurringBillResponse(
			RecurringBillResponse lastRecurringBillResponse) {
		this.lastRecurringBillResponse = lastRecurringBillResponse;
	}
	public Object get_links() {
		return _links;
	}
	public void set_links(Object _links) {
		this._links = _links;
	}
	public List<Errors> getErrors() {
		return errors;
	}
	public void setErrors(List<Errors> errors) {
		this.errors = errors;
	}
	public String getRowData() {
		return rowData;
	}
	public void setRowData(String rowData) {
		this.rowData = rowData;
	}
	/**
	 * @param transactionManager the transactionManager to set
	 */
	public void setTransactionManager(TransactionManager transactionManager) {
		this.transactionManager = transactionManager;
	}
	/**
	 * @return the billInformation
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	public BillInformation getBillInformation() throws JsonParseException, JsonMappingException, IOException {
		if(billInformation==null){
			BillInformation b = new BillInformation(this.transactionManager);
			b.setUrl(this.transactionManager.getUrl()+"recurring-bill/");
			b.getBillForRecurringBillInformationByTransactionId(lastRecurringBillResponse.getRecurringBillId());
			billInformation=b;
		}
		return billInformation;
	}
	/**
	 * @return the cardDataInformation
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	public CardDataInformation getCardDataInformation() throws JsonParseException, JsonMappingException, IOException {
		if(cardDataInformation==null){
			CardDataInformation c = new CardDataInformation(this.transactionManager);
			c.getDataByTransaction(TransactionType.RecurringBill, lastRecurringBillResponse.getRecurringBillId());
			cardDataInformation=c;			
			}
		return cardDataInformation;
	}

	/**
	 * @return the customerInformation
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	public CustomerInformation getCustomerInformation() throws JsonParseException, JsonMappingException, IOException {
		if(customerInformation==null){
				CustomerInformation c = new CustomerInformation(this.transactionManager);
				c.setUrl(this.transactionManager.getUrl()+"recurring-bill/");
				c.getCustomerForSaleInformationByTransactionId(lastRecurringBillResponse.getRecurringBillId());
				customerInformation=c;				
			}
		return customerInformation; 
	}

	/**
	 * @return the merchantInformation
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	public MerchantInformation getMerchantInformation() throws JsonParseException, JsonMappingException, IOException {
		if(merchantInformation==null){			
				MerchantInformation m = new MerchantInformation(this.transactionManager);
				m.getDataByTransaction(TransactionType.RecurringBill, lastRecurringBillResponse.getRecurringBillId());
				merchantInformation=m;				
			}
		return merchantInformation;
	}
	/**
	 * @return the scheduleInformation
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	public ScheduleInformation getScheduleInformation() throws JsonParseException, JsonMappingException, IOException {
		if(scheduleInformation==null){			
			ScheduleInformation s = new ScheduleInformation(this.transactionManager);
			s.getDataByTransaction(TransactionType.Schedule, lastRecurringBillResponse.getRecurringBillId());
			scheduleInformation=s;				
		}
		return scheduleInformation;
	}
	public StatusInformation getStatusInformation() throws JsonParseException, JsonMappingException, IOException {
		if(status==null){			
			StatusInformation s = new StatusInformation(this.transactionManager);
			s.getDataByTransaction(TransactionType.RecurringBill, lastRecurringBillResponse.getRecurringBillId());
			status=s;				
		}
		return status;
	}

}
